org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -16,63 +16,101 @@ namespace MixedReality.Toolkit.Input
     [AddComponentMenu("MRTK/Input/Near Interaction Mode Detector")]
     public class NearInteractionModeDetector : ProximityDetector
     {
+        /// <summary>
+        /// The set of near interactables that belongs to near interaction
+        /// </summary>
         [SerializeField]
-        [Tooltip("The set of near interactors that belongs to near interaction")]
-        private List<XRBaseInteractor> nearInteractors;
+        [Tooltip("The set of near interactables that belongs to near interaction")]
+        private List<XRBaseInteractable> nearInteractables;
 
         /// <summary>
-        /// Used to keep track of the previously detected colliders so that we can know which
-        /// colliders stopped being detected and update their buttons front plate and rounded
-        /// rect if they are labeled as dynamic (based on proximity).
+        /// Used to keep track of the previously detected colliders+interactrables duples (CID) so that we can know which
+        /// CID stopped being detected and update their buttons front plate RawImage.
         /// </summary>
-        List<Collider> previouslyDetectedColliders = new List<Collider>();
+        private HashSet<(Collider, IXRProximityInteractable)> previouslyDetectedColliderInteractableDuples = new HashSet<(Collider, IXRProximityInteractable)>();
```

Comment:
Why are you storing both Collider and IXRProximityInteractable?

You should be able to do what you need by only storing IXRProximityInteractable.   Perhaps we should chat.

Reply:
That's a remnant from the initial approach, I was under the impression that it was needed to uniquely identify which UIElement was receiving the hoverProximity event and when trying to access both the FrontPlate RawImage and the Canvas' CanvasRoundedRect, now I realize it is indeed not needed for just modifying the FrontPlate's RawImage.  I've removed it in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/659/commits/460c96e37cfc4d3ad2b7bfb54cdc14d4e9ed733d